### PR TITLE
Add fastremap-bio

### PR DESCRIPTION
**Pull request by canfirtina**
Adding the FastRemap tool to bioconda.

Original GitHub page: https://github.com/CMU-SAFARI/FastRemap
arXiv link describing the paper: https://arxiv.org/pdf/2201.06255.pdf

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by canfirtina** [comment]
@BiocondaBot please add label

File: recipes/fastremap-bioconda/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{% set version = "1.0.0" %}
+
+package:
+  name: fastremap-bioconda
+  version: 1.0.0
+
+source:
+  url: https://github.com/CMU-SAFARI/FastRemap/releases/download/v1.0.0/FastRemap-1.0.0.tar.gz
```
Comment (cbrueffer) [inline recipes/fastremap-bioconda/meta.yaml:8]:
```suggestion
  url: https://github.com/CMU-SAFARI/FastRemap/releases/download/v{{ version }}/FastRemap-{{ version }}.tar.gz
```

File: recipes/fastremap-bioconda/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  sha256: 269e4a95b1183ae151ae94c8c4c4d3b45de174a72dc21c31dc384e62164ec26b
+
+build:
+  number: 0
+  skip: True  # [osx]
+
+requirements:
+  build:
+    - {{ compiler('gxx') }}
```
Comment (cbrueffer) [inline recipes/fastremap-bioconda/meta.yaml:17]:
```suggestion
    - {{ compiler('cxx') }}
```

File: recipes/fastremap-bioconda/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{% set version = "1.0.0" %}
+
+package:
+  name: fastremap-bioconda
```
Comment (cbrueffer) [inline recipes/fastremap-bioconda/meta.yaml:4]:
```suggestion
  name: fastremap
```

**Review by cbrueffer — COMMENTED** [review]
Looks OK, but please rename the recipe to `fastremap`; there's no need (and no precedent) for the suffix.

**Comment by canfirtina** [comment]
@cbrueffer Sorry, I closed the pull request by mistake while trying to rename my branch to `fastremap`, which was probably unnecessary. I am reopening now. I applied the changes you suggested.

It is OK for us to change the recipe name to `fastremap` from `fastremap-bioconda`. We used that naming only because we noticed the following package: https://anaconda.org/conda-forge/fastremap.

**Comment by canfirtina** [comment]
We definitely would like to apply the changes based on your insight and opinion, @cbrueffer. We will use`fastremap-bio` instead of `fastremap`. I will make the changes accordingly.

**Comment by cbrueffer** [comment]
Just to clarify, what I meant was not only renaming the recipe here, but renaming your software completely. Two pieces of software with the same name in essentially the same problem space is not good, particularly when it's easy to avoid the issue given the knowledge about it now.

**Comment by canfirtina** [comment]
Oh I see. Thanks for the clarification. We will keep the tool name as is for now, and use `fastremap-bio` as our recipe name. The test should finish successfully in a moment. Could you please review and merge if everything looks good?

**Comment by canfirtina** [comment]
Hi @cbrueffer, this is a reminder in case you missed my latest message because I did not tag you there.

**Comment by daler** [comment]
Just chiming in here, I agree with @cbrueffer that packages with the same name are asking for confusion.  We're just trying to save you and your users from problems in the future. It doesn't look like your executables clobber those of the existing fastremap, but you may want to coordinate future updates with the authors of that package to ensure your packages can be installed in the same environment.

Also, I noticed that [the tarball on github](https://github.com/CMU-SAFARI/FastRemap/releases/download/v1.0.0/FastRemap-1.0.0.tar.gz) is 160 MB, almost all of which appears to be in a hidden `.git` directory in the tarball. Can you either remove that from the release (preferable), or delete that directory from within the install script here?

**Comment by canfirtina** [comment]
Thanks for the suggestions, @daler. Removing the hidden `.git` directory helped to significantly reduce the released tarball size on GitHub.

I will be in contact with the authors of the `fastremap` package in case we encounter issues with installing both `fastremap` and `fastremap-bio` in the same environment.

Please let me know if I need to make any other changes.

**Comment by canfirtina** [comment]
Bumping this up @cbrueffer @daler. Is there any prediction regarding when `fastremap-bio` would be available in the bioconda channel?

Thanks,
Can

**Comment by dpryan79** [comment]
@BiocondaBot please fetch artifacts

**Comment by dpryan79** [comment]
@BiocondaBot please fetch artifacts

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@BiocondaBot please merge

**Pull request merged**